samza-api/src/main/java/org/apache/samza/system/IncomingMessageEnvelope.java

```diff
@@ -37,10 +38,20 @@
   private final Object key;
   private final Object message;
   private final int size;
-  // the System's nanoTime when this event occurred, should be set by the event source, 0 means unassigned
-  private long eventTime = 0L;
-  // the System's nanoTime when this event is picked-up by samza, 0 means unassgined
-  private long arrivalTime;
+
+  /**
+   * the Instant when this event occurred, should be set by the event source,
+   * default null if not available or unassigned
```

Comment:
Instead of needing to deal with `null`, maybe we can just default to `Instant.ofEpochMilli(0)`? That is what is treated as the default anyways for `getEventTime`.

Reply:
makes sense. done.